contracts/staking/PoSValidatorManager.sol

```diff
@@ -158,29 +156,32 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         require(
             validator.status == ValidatorStatus.Active, "PoSValidatorManager: validator not active"
         );
+        // Update the validator weight
+        uint64 newValidatorWeight = validator.weight + weight;
+        _setValidatorWeight(validationID, newValidatorWeight);
+
+        uint64 nonce = _getAndIncrementNonce(validationID);
+        bytes32 delegationID = sha256(abi.encodePacked(validationID, delegatorAddress, nonce));
 
-        // Ensure the delegator is not already registered
+        // Ensure the delegationID is not already registered
         require(
-            $._delegatorStakes[validationID][delegator].weight == 0,
-            "PoSValidatorManager: delegator already registered"
+            $._delegatorStakes[delegationID].weight == 0,
```

Comment:
I think this check can be removed since `delegationID` is guaranteed to be unique for every call right?

Reply:
True, nice catch